File: src/ua/edu/hneu/oop/java/lr3/task3/Hyperbola.java
package ua.edu.hneu.oop.java.lr3.task3;

import java.util.HashMap;
import java.util.Map;

public class Hyperbola extends Function {
    private double a, b;

    public Hyperbola(double a, double b) {
        this.a = a;
        this.b = b;
    }

    @Override
    public double calculate(double x) throws ArithmeticException {
        if (x == 0) {
            throw new ArithmeticException("division by zero");
        }
        return a / x + b;
    }

    @Override
    public Map<String, Double> getParameters() {
        Map<String, Double> params = new HashMap<>();
        params.put("a", a);
        params.put("b", b);
        return params;
    }

    @Override
    public String toString() {
        return "(a/x) + b";
    }

    public double getA() {
        return a;
    }

    public void setA(double a) {
        this.a = a;
    }

    public double getB() {
        return b;
    }

    public void setB(double b) {
        this.b = b;
    }
}

File: src/ua/edu/hneu/oop/dotNet/LR3/Task3/Hyperbola.cs
using System;
using System.Collections.Generic;

namespace Task3
{
    public class Hyperbola : Function
    {
        public Hyperbola(double a, double b)
        {
            A = a;
            B = b;
        }

        public double A { get; }
        public double B { get; }


        public override double Calculate(double x)
        {
            if (x == 0)
                throw new ArithmeticException("division by zero");

            return A / x + B;
        }

        public override Dictionary<String, Double> GetParameters()
        {
            return new Dictionary<String, Double> {{"a", A}, {"b", B}};
        }

        public override String ToString()
        {
            return "(a/x) + b";
        }
    }
}